.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,116 @@
+name: Docker Build, Publish and Deploy
+
+on:
+  push:
+    branches: [master]
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  docker_build_and_publish:
+    runs-on: ubuntu-latest
+    outputs:
+      IMAGE_TAGE: ${{ steps.image_tag.outputs.IMAGE_TAG }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      - name: Define_docker_image_tag
+        run: |
+          echo "DOCKER_IMAGE_TAG=(git describe --tags)" >> $GITHUB_ENV
+          echo "IMAGE_TAG=$(git describe --tags)" >> "$GITHUB_OUTPUT"
+      
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with: 
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Build and Push
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: 'linux/amd64'
+          push: true
+          tags: nethermindeth/juno:${{ github.event.inputs.tag }}
```

Comment:
There is no `github.event.inputs.tag` defined, it won't work. Shouldn't be `IMAGE_TAG ` here?